### PR TITLE
fix: [PIE-4751]: add tests and do polling only after resolving last call

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary

- correct the test for polling
- do polling only after resolving the last call.

Previously: make calls every 5 seconds
Now: Set the timeout for the next call only after the current call is resolved.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/pages/execution-list/ExecutionList.tsx [hunk 1]
```diff
@@ -106,8 +105,9 @@ function ExecutionListInternal(props: ExecutionListProps): React.ReactElement {
           filterType: 'PipelineExecution'
         }
   })
-  const tabVisible = useTabVisible()
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/execution-list/ExecutionList.tsx:109]: moved the tabVisible logic to usePolling to make it reusable across.

File: src/modules/70-pipeline/hooks/usePolling.ts [hunk 2]
```diff
@@ -6,35 +6,39 @@
...
     }
-    const timerId = setTimeout(() => {
-      setIsPolling(true)
-      savedCallback.current()?.finally(() => setIsPolling(false))
-    }, POLLING_INTERVAL_IN_MS)
 
-    return () => clearTimeout(timerId)
-  }, [isPolling, startPolling])
+    // Poll only when the current request is resolved
+    if (!isPolling) {
```
Comment (udayvunnam) [inline src/modules/70-pipeline/hooks/usePolling.ts:31]: effect runs but the timeout is started only after last call is resolved

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 2 Issues 
* Code issue 2 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10461)

File: src/modules/70-pipeline/pages/execution-list/__tests__/ExecutionList.test.tsx [hunk 3]
```diff
@@ -404,42 +404,24 @@ describe('Test Pipeline Deployment list', () => {
...
-        status: [],
-        filterIdentifier: undefined,
-        module: 'cd',
-        myDeployments: false,
-        repoIdentifier: undefined,
-        searchTerm: undefined,
-        size: 20,
-        branch: undefined
-      }
-    })
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/execution-list/__tests__/ExecutionList.test.tsx:442]: This test was not doing an assertion for polling as the title says earlier

**Review by vkbansal-harness — APPROVED** [review]

**Comment by udayvunnam** [comment]
retrigger jest

**Comment by udayvunnam** [comment]
trigger cypress

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**